docs/source/examples.md

```diff
@@ -1,5 +1,163 @@
+
+```bash
+cd examples/1
+python sample.py -n 103
+```
+
+The sample [examples/2/sample.py](examples/2/sample.py) shows the minimal modifications to run this in AzureML. Firstly create an AzureML workspace and download the config file, as explained [here](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-configure-environment). The config file should be placed in the same folder as the sample script. A sample [Conda environment file](examples/2/environment.yml) is supplied. Import the [hi-ml package](https://pypi.org/project/hi-ml/) into the current environment. Finally add the following to the sample script:
+
+```python
+from health.azure.himl import submit_to_azure_if_needed, WORKSPACE_CONFIG_JSON
+```
+
+and add the following at the beginning of main:
+
+```python
+    _ = submit_to_azure_if_needed(
+        compute_cluster_name="lite-testing-ds2",
+        workspace_config_path=WORKSPACE_CONFIG_JSON,
+        conda_environment_file=Path("environment.yml"),
+        wait_for_completion=True,
+        wait_for_completion_show_output=True)
+```
+
+If this script is invoked as the first sample, e.g.
+
+```bash
+cd examples/2
+python sample.py -n 103
+```
+
+then the output will be exactly the same. But if the script is invoked as follows:
+
+```bash
+cd examples/2
+python sample.py -n 103 --azureml
+```
+
+then the function `submit_to_azure_if_needed` will do all the required actions to run this script in AzureML and exit. Note that:
+
+* code after `submit_to_azure_if_needed` is not run.
+* the print statement prints to the AzureML console output and is available in the `Output + logs` tab of the experiment in the `70_driver_log.txt` file, and can be downloaded from there.
+* the command line arguments are passed through (apart from --azureml) when running in AzureML.
+* a new file: `most_recent_run.txt` will be created containing an identifier of this AzureML run.
+
+A sample script [examples/2/results.py](examples/2/results.py) demonstrates how to programmatically download the driver log file.
+
+## Output files
+
+The sample [examples/3/sample.py](examples/3/sample.py) demonstrates output file handling when running on AzureML. Because each run is performed in a separate VM or cluster any file output is not generally preserved. In order to keep the output it should be written to the `outputs` folder when running in AzureML. The AzureML infrastructure will preserve this and it will be available for download from the `outputs` folder in the `Output + logs` tab.
+
+Make the following additions:
+
+```python
+    run_info = submit_to_azure_if_needed(
+
+    ...
+
+    parser.add_argument("-o", "--output", type=str, default="primes.txt", required=False, help="Output file name")
+
+    ...
+
+    run_info.output_folder.mkdir(exist_ok=True)
```

Comment:
this mkdir should actually happen inside of our package!

Reply:
Added mkdirs for logs and outputs